crates/sui-core/src/rpc_index.rs

```diff
@@ -348,11 +370,48 @@ struct IndexStoreTables {
     /// Allows efficient listing of all versions of a package.
     #[default_options_override_fn = "default_table_options"]
     package_version: DBMap<PackageVersionKey, PackageVersionInfo>,
+
+    /// Authenticated events index by (stream_id, checkpoint_seq, transaction_idx, event_index)
+    /// Value is the full sui_types::event::Event
+    events_by_stream: DBMap<EventIndexKey, Event>,
```

Comment:
Can we avoid storing the event here and instead fetch it from the perpetual store? This is slightly more overhead for a dramatically reduced storage footprint. essentially we can have the value be `()`.